packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -221,7 +221,11 @@ class AnsiTerminal {
   String clearScreen() => supportsColor ? _clear : '\n\n';
 
   set singleCharMode(bool value) {
-    stdin.lineMode = !value;
+    try {
+      stdin.lineMode = !value;
+    } on IOException {
+      // This can throw for some terminals; we ignore the error.
```

Comment:
So as far as I can tell this error (ENOTTY) can only occur when the terminal is not a terminal (like, it's a redirect). Is that when we're seeing it?

Do we have enough data from the exception to identify that this is the error we're getting?